blocks/editable/index.js

```diff
@@ -108,7 +108,11 @@ export default class Editable extends wp.element.Component {
 		const collapsed = this.editor.selection.isCollapsed();
 
 		this.setState( {
-			empty: ! content || ! content.length,
+			empty:
+				! content ||
+				! content.length ||
+				// On non-inline editables, TinyMCE appends an empty <p> tag
+				( ! this.props.inline && content.length === 1 && ! content[ 0 ].props.children ),
```

Comment:
Could we hook into TinyMCE's internal empty checking?

https://github.com/tinymce/tinymce/blob/master/src/core/src/main/js/dom/Empty.js

I was wondering why we wouldn't have had to account for `br[data-mce-bogus]` which is added similar to empty `<p>` tag here. Appears we do some clean-up in `getContent` already. Should we be stripping the empty paragraphs there?